docs: Updates for 3.0 and more (#2941)
looker-open-source/components: www/src/documentation/develop/theme.mdx

}
```

## Customizing the Theme

> You can extend the theme provided by Looker Components by passing colors, fonts, and more into the `themeCustomizations` prop.

Reviewer: nit: Might be nice to remove the word "extend" from the opening sentence. Happy to keep it around though if preferable.
```suggestion
You can customize the theme provided by Looker Components by passing colors, fonts, and more into the `themeCustomizations` prop.
```

```jsx
;() => {
  /**
   *  import { Button, ComponentsProvider } from '@looker/components'
   **/

  return (
    <ComponentsProvider themeCustomizations={{ colors: { key: 'hotpink' } }}>
      <Button>Pink is so Pretty!</Button>
    </ComponentsProvider>
  )
}
```

> The following colors are available for customization, the first three being the most likely to need to be customized:

Reviewer: Super nit: Could be useful to give some context as to why we think these colors require customizing.
```suggestion
The following colors are available for customization. The first three are most likely to require customization since they are heavily used throughout the Looker Components library:
```

- key: Key color is applied strategically across the UI. Used for default action buttons, toggle switches, interactive component accents

> - text: Used for text

Reviewer 2: The background specifies that it's a default. I think that makes sense for the text color as well. Something like "Default text color" is short and descriptive.

Author: ✅

- background: Default page background used for application background

> These colors are less likely to need to be customized:

Reviewer: Makes this line a little more consistent with my last suggestion. Feel free to keep as is.
```suggestion
These colors are less likely to require customization:
```

> - pageBackground: Used as page background in some themes pageBackground and background have a subtle difference, wherein the pageBackground is closer to `ui1` but blends are calculated from a more "absolute" background color (white or black)

Reviewer: ```suggestion
- pageBackground: Used as page background in some themes. pageBackground and background have a subtle difference, wherein the pageBackground is closer to `ui1` but blends are calculated from a more "absolute" background color (white or black)
```

- link: Link text color, used for Breadcrumb, link
- critical: Critical intent color used for: Delete button, error and validation messages
- warn: Warn intent color used for Warning banner
- positive: Positive intent color used for Positive banner
- inform: Inform intent color used for Info banner
- calculation: Indicates a calculated value
- dimension: Indicates dimensionality
- measure: Indicates a measure
- title: Used by Heading. This is generally identical to colors.text5
- body: Used by Paragraph, Ul/Ol. This is generally identical to colors.text5

The following font family options may be customized (use the `fontSources` property to specify the sources):

- body: Used for body text
- brand: Used for headings
- code: Used for code snippets

> Additionaly properties to customize:

Reviewer: ```suggestion
Additional properties to customize:
```

- brandAnimation: The Google Material ripple effect for pressable elements
- externalLabel: True sets the field label style to external, false sets it to the Google Material floating label style

## Nesting ComponentsProvider (ThemeProviders)

> `ExtendComponentsThemeProvider` as a Styled Components' ThemeProvider acts as a React Context and in so doing is able to support the standard behavior of nested contexts - the theme defined nearest to the component will be the version used.

Reviewer: I _think_ that's supposed to be an "is" but I could be totally off. Wanted to point it out.
```suggestion
`ExtendComponentsThemeProvider` is a Styled Components' ThemeProvider that acts as a React Context and in so doing is able to support the standard behavior of nested contexts - the theme defined nearest to the component will be the version used.
```
